internal/rbd/rbd_util.go

```diff
@@ -1934,13 +1945,12 @@ func (rv *rbdVolume) setVolumeMetadata(parameters map[string]string) error {
 	return nil
 }
 
-// setSnapshotMetadata Set snapshot-name/snapshot-namespace/snapshotcontent-name metadata
-// on RBD image.
-func (rv *rbdVolume) setSnapshotMetadata(parameters map[string]string) error {
-	for k, v := range k8s.GetSnapshotMetadata(parameters) {
-		err := rv.SetMetadata(k, v)
-		if err != nil {
-			return fmt.Errorf("failed to set metadata key %q, value %q on image: %w", k, v, err)
+// unsetAllMetadata unset all the metadata from arg keys on RBD image.
+func (rv *rbdVolume) unsetAllMetadata(keys []string) error {
+	for _, key := range keys {
+		err := rv.RemoveMetadata(key)
+		if err != nil && !strings.Contains(err.Error(), "No such file or directory") {
+			return fmt.Errorf("failed to unset metadata key %q on %q: %w", key, rv, err)
```

Comment:
```suggestion
if err != nil && !errors.Is(err, rados.ErrNotFound) {
return fmt.Errorf("failed to unset metadata key %q on %q: %w", key, rv, err)
}
```

can we do like this?

Reply:
@Madhu-1 This is what I'm talking about in the call, can I revert this?
rbd upgrade test is failing with:
```
May 10 13:01:46.985: INFO: At 2022-05-10 12:51:48 +0100 BST - event for rbd-pvc-restore: {rbd.csi.ceph.com_csi-rbdplugin-provisioner-57c4f55c9c-thjpc_96186737-0ddd-4b5f-bef5-2df83e7ac0d1 } ProvisioningFailed: failed to provision volume with StorageClass "csi-rbd-sc": rpc error: code = Unknown desc = failed to unset metadata key "csi.storage.k8s.io/volumesnapshot/name" on "replicapool/csi-vol-92685dd1-d057-11ec-83c4-862ba5925cfb": rbd: ret=-2, No such file or directory
```